tests/Core/Content/Product/SalesChannel/Listing/ProductListingFeaturesSubscriberTest.php

```diff
@@ -1042,4 +1057,63 @@ public function testCriteriaLimitIsSetForPagination(
         $subscriber = $this->getDefaultProductListingFeaturesSubscriber();
         $subscriber->{$endpoint}($eventMock);
     }
+
+    public function testListingRequestIsNotHandledWhenDeepCategoryIsMainCategory(): void
+    {
+        $salesChannelContextMock = $this->getMockBuilder(SalesChannelContext::class)
+            ->disableOriginalConstructor()
+            ->getMock();
+
+        /** @var SalesChannelEntity|MockObject $salesChannelMock */
+        $salesChannelMock = $this->getMockBuilder(SalesChannelEntity::class)
+            ->disableOriginalConstructor()
+            ->getMock();
+
+        $navigationCategoryMock = $this->getMockBuilder(CategoryEntity::class)
+            ->disableOriginalConstructor()
+            ->getMock();
+
+        $navigationCategoryMock->expects($this->any())->method('getId')->willReturn(Uuid::randomHex());
+
+        $navigationCategoryMock->expects($this->any())->method('getBreadcrumb')
+            ->willReturn(['FINDOLOGIC Main 2', 'FINDOLOGIC Sub', 'Very deep']);
+
+        $salesChannelMock->expects($this->once())->method('getNavigationCategory')
+            ->willReturn($navigationCategoryMock);
```

Comment:
Does that mean `'FINDOLOGIC Main 2/FINDOLOGIC Sub/Very deep'` is the main/root category? It's not visible at first sight.

Reply:
Yes. Not sure how I should improve on that, maybe you have some ideas?